lib/std/json.zig

```diff
@@ -2648,17 +2673,9 @@ pub fn stringify(
             }
         },
         .Enum => {
-            if (comptime std.meta.trait.hasFn("jsonStringify")(T)) {
```

Comment:
only enum/union/struct can have user traits; it makes more sense to keep them here?

Reply:
This is exactly why I found it reasonable to move it outside: it stops this check from getting repeated in the code without altering behavior. Since I applied the same reasoning to `jsonParse` and `jsonStringify` is effectively its pair I thought it was a good idea to have the same logic applied to it.